tests/unit/test_bruteforce_multi.cpp

```diff
@@ -179,251 +179,296 @@ TEST_F(BruteForceMultiTest, brute_force_empty_index) {
-//     }
-//     ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+    float query[] = {0, 0, 0, 0};
+    auto verify_res = [&](size_t id, float score, size_t index) {
+        ASSERT_EQ(id, n_labels - index - 1);
+    };
+    // Having k = n_labels - 1, the heap will continuously pop the worst label before finding the
+    // next vector, which is of the same label and is the best vector yet.
+    runTopKSearchTest(index, query, n_labels - 1, verify_res);
 
-//     // Run the same query again.
-//     runTopKSearchTest(index, query, k, verify_res);
+    // Having k = n_labels, the heap will never get to pop the worst label, but we will update the
+    // scores each time.
+    runTopKSearchTest(index, query, n_labels, verify_res);
 
-//     VecSimIndex_Free(index);
-// }
+    VecSimIndex_Free(index);
+}
 
-// TEST_F(BruteForceMultiTest, brute_force_reindexing_same_vector_different_id) {
-//     size_t n = 100;
-//     size_t k = 10;
-//     size_t dim = 4;
+TEST_F(BruteForceMultiTest, reindexing_same_vector_different_id) {
+    size_t n = 100;
+    size_t k = 10;
+    size_t dim = 4;
+    size_t perLabel = 3;
 
-//     VecSimParams params{.algo = VecSimAlgo_BF,
-//                         .bfParams = BFParams{.type = VecSimType_FLOAT32,
-//                                              .dim = dim,
-//                                              .metric = VecSimMetric_L2,
-//                                              .multi = true,
-//                                              .initialCapacity = 200}};
-//     VecSimIndex *index = VecSimIndex_New(&params);
+    VecSimParams params{.algo = VecSimAlgo_BF,
+                        .bfParams = BFParams{.type = VecSimType_FLOAT32,
+                                             .dim = dim,
+                                             .metric = VecSimMetric_L2,
+                                             .multi = true,
+                                             .initialCapacity = 200}};
+    VecSimIndex *index = VecSimIndex_New(&params);
 
-//     for (size_t i = 0; i < n; i++) {
-//         float f[dim];
-//         for (size_t j = 0; j < dim; j++) {
-//             f[j] = (float)(i / 10); // i / 10 is in integer (take the "floor" value)
-//         }
-//         VecSimIndex_AddVector(index, (const void *)f, i);
-//     }
-//     ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+    for (size_t i = 0; i < n; i++) {
+        float f[dim];
+        for (size_t j = 0; j < dim; j++) {
+            f[j] = (float)(i / 10); // i / 10 is in integer (take the "floor" value)
+        }
+        VecSimIndex_AddVector(index, (const void *)f, i);
+    }
+    // Add more vectors under the same labels. their scores should be worst.
+    for (size_t i = 0; i < n; i++) {
+        float f[dim];
+        for (size_t j = 0; j < dim; j++) {
+            f[j] = (float)(i / 10) + n;
+        }
+        for (size_t j = 0; j < perLabel - 1; j++) {
+            VecSimIndex_AddVector(index, (const void *)f, i);
```

Comment:
You add here the same vector 3 times for label `i`? is that what you meant?

Reply:
Yes. This is a test from the BF tests, and I simply want to check that it passes on multi-value as well.